pkg/validate/validate_test.go

```diff
@@ -0,0 +1,185 @@
+package validate
+
+import (
+	"testing"
+
+	"github.com/Azure/acs-engine/pkg/api/osa/vlabs"
+)
+
+/*
+name: openshift
+location: eastus
+properties:
+  openShiftVersion: "$DEPLOY_VERSION"
+  publicHostname: openshift.$RESOURCEGROUP.$DNS_DOMAIN
+  routerProfiles:
+  - name: default
+    publicSubdomain: $RESOURCEGROUP.$DNS_DOMAIN
+  agentPoolProfiles:
+  - name: master
+    role: master
+    count: 3
+    vmSize: Standard_D2s_v3
+    osType: Linux
+  - name: infra
+    role: infra
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+  - name: compute
+    role: compute
+    count: 1
+    vmSize: Standard_D2s_v3
+    osType: Linux
+  servicePrincipalProfile:
+    clientID: $AZURE_CLIENT_ID
+    secret: $AZURE_CLIENT_SECRET
+*/
+func TestValidate(t *testing.T) {
+
+	tests := map[string]struct {
+		oc            *vlabs.OpenShiftCluster
+		expectedError string
+	}{
+		"invalid version": {
+			oc: &vlabs.OpenShiftCluster{
```

Comment:
turn this into a `getValidOpenShiftCluster` and use the closure to make the version invalid